src/accessControl/acp/index.js

```diff
@@ -201,18 +201,35 @@ function ensureAccessControl(policyUrl, datasetWithAcr, changed) {
 }
 
 function ensureApplyControl(policyUrl, datasetWithAcr, changed) {
-  const accessControls = acp.getAllControl(datasetWithAcr);
+  let accessControls = [];
+  try {
+    accessControls = acp.getAllControl(datasetWithAcr);
+  } catch (error) {
+    // TODO: Handle this error
+    // It doesn't seem to affect the outcome though, so we'll leave it like this for now
+  }
   const existingAccessControl = accessControls.find((ac) =>
     acp.getPolicyUrlAll(ac).find((url) => policyUrl === url)
   );
+  if (existingAccessControl) {
+    return {
+      changed,
+      acr: datasetWithAcr,
+    };
+  }
+  let acr = datasetWithAcr;
+  try {
+    acr = acp.setControl(
+      datasetWithAcr,
+      chain(acp.createControl(), (ac) => acp.addPolicyUrl(ac, policyUrl))
+    );
+  } catch (error) {
+    // TODO: Handle this error
+    // It doesn't seem to affect the outcome though, so we'll leave it like this for now
```

Comment:
Same here - how does `setControl` failing not affect the outcome? What are the possible errors?

Reply:
Same answer as above.

I could debug further to figure out exactly what's going on, but I'm not sure it's worth the time it will take.